test/stream-for-each.cc

```diff
@@ -0,0 +1,271 @@
+  ASSERT_FALSE(interrupted.load());
+
+  k.Start();
+
+  ASSERT_TRUE(waiting.load());
+  ASSERT_FALSE(interrupted.load());
+
+  interrupt.Trigger();
+
+  EXPECT_THROW(future.get(), stout::eventuals::StoppedException);
+
+  ASSERT_TRUE(interrupted.load());
+}
+
+// Iterate with init_list
+TEST(StreamForEach, Let) {
+  auto s = []() {
+    return Iterate(std::vector<int>{1, 2})
+        | StreamForEach(Let([](int& x) {
+             return Iterate(std::vector<int>{1, 2})
+                 | StreamForEach(Let([&x](int& y) {
+                      return Iterate(std::vector<int>{x, y});
+                    }));
+           }))
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_THAT(*s(), ElementsAre(1, 1, 1, 2, 2, 1, 2, 2));
+}
+
+TEST(StreamForEach, TwoLevelLoop) {
+  auto s = []() {
+    return Range(2)
+        | StreamForEach([](int x) { return Range(2); })
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_THAT(*s(), ElementsAre(0, 1, 0, 1));
+}
+
+TEST(StreamForEach, TwoIndexesSum) {
+  auto s = []() {
+    return Range(3)
+        | StreamForEach([](int x) {
+             return Stream<int>()
+                 .next([container = std::vector<int>({1, 2}),
+                        i = 0u,
+                        x](auto& k) mutable {
+                   if (i != container.size()) {
+                     k.Emit(container[i++] + x);
+                   } else {
+                     k.Ended();
+                   }
+                 })
+                 .done([](auto& k) {
+                   k.Ended();
+                 });
+           })
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_THAT(*s(), ElementsAre(1, 2, 2, 3, 3, 4));
+}
+
+TEST(StreamForEach, TwoIndexesSumMap) {
+  auto s = []() {
+    return Range(3)
+        | StreamForEach([](int x) { return Range(1, 3)
+                                        | Map(Then([x](int y) { return x + y; })); })
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_THAT(*s(), ElementsAre(1, 2, 2, 3, 3, 4));
+}
+
+TEST(StreamForEach, ThreeIndexesSumMap) {
+  auto s = []() {
+    return Range(3)
+        | StreamForEach([](int x) { return Range(1, 3)
+                                        | Map(Then([x](int y) { return x + y; })); })
+        | StreamForEach([](int x) { return Range(1, 3)
+                                        | Map(Then([x](int y) { return x + y; })); })
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_THAT(*s(), ElementsAre(2, 3, 3, 4, 3, 4, 4, 5, 4, 5, 5, 6));
+}
+
+TEST(StreamForEach, ThreeLevelLoop) {
+  auto s = []() {
+    return Range(2)
+        | StreamForEach([](int x) { return Range(2); })
+        | StreamForEach([](int x) { return Range(2); })
+        | Collect<std::vector<int>>();
+  };
+
+  EXPECT_THAT(*s(), ElementsAre(0, 1, 0, 1, 0, 1, 0, 1));
+}
+
+TEST(StreamForEach, ThreeLevelLoop2) {
+  auto s = []() {
+    return Range(2)
+        | StreamForEach([](int x) { return Range(2) | StreamForEach([](int y) { return Range(2); }); })
```

Comment:
How did you get this past the column length checker? Are you using the pre-commit hook?

Again, let's wrap this at the first `return` and then at subsequent `|` so it's easier to read, thanks!